packages/@uppy/image-generator/src/index.tsx

```diff
@@ -0,0 +1,295 @@
+import type {
+  Body,
+  Meta,
+  MinimalRequiredUppyFile,
+  UIPluginOptions,
+  UppyEventMap,
+} from '@uppy/core'
+import Uppy, { UIPlugin } from '@uppy/core'
+import { FilterInput, SearchView } from '@uppy/provider-views'
+import type { AssemblyOptions, AssemblyResult } from '@uppy/transloadit'
+import Transloadit from '@uppy/transloadit'
+import locale from './locale.js'
+
+export interface ImageGeneratorOptions extends UIPluginOptions {
+  assemblyOptions: (prompt: string) => Promise<AssemblyOptions>
+}
+
+interface PluginState extends Record<string, unknown> {
+  prompt: string
+  results: AssemblyResult[]
+  checkedResultIds: Set<AssemblyResult['id']>
+  loading: boolean
+  loadingMessageIndex: number
+  firstRun: boolean
+}
+
+const defaultState = {
+  prompt: '',
+  results: [],
+  checkedResultIds: new Set(),
+  loading: false,
+  loadingMessageIndex: 0,
+  firstRun: true,
+} satisfies PluginState
+
+const LOADING_MESSAGES = [
+  'generating1',
+  'generating2',
+  'generating3',
+  'generating4',
+  'generating5',
+] as const
+
+export default class ImageGenerator<
+  M extends Meta,
+  B extends Body,
+> extends UIPlugin<ImageGeneratorOptions, M, B, PluginState> {
+  private loadingInterval: ReturnType<typeof setInterval> | null = null
+  private localUppy?: Uppy<M, B>
+
+  constructor(uppy: Uppy<M, B>, opts: ImageGeneratorOptions) {
+    super(uppy, opts)
+
+    this.id = this.opts.id || 'ImageGenerator'
+    this.title = 'AI image'
+    this.type = 'acquirer'
+
+    this.defaultLocale = locale
+
+    this.setPluginState(defaultState)
+
+    this.i18nInit()
+  }
+
+  install(): void {
+    const { target } = this.opts
+    if (target) {
+      this.mount(target, this)
+    }
+  }
+
+  uninstall(): void {
+    this.clearLoadingInterval()
+    this?.localUppy?.destroy()
+    this.localUppy = undefined
+    this.unmount()
+  }
+
+  private clearLoadingInterval(): void {
+    if (this.loadingInterval) {
+      clearInterval(this.loadingInterval)
+      this.loadingInterval = null
+    }
+  }
+
+  private startLoadingAnimation(): void {
+    this.clearLoadingInterval()
+    this.loadingInterval = setInterval(() => {
+      const { loadingMessageIndex } = this.getPluginState()
+      const nextIndex = (loadingMessageIndex + 1) % LOADING_MESSAGES.length
+      this.setPluginState({ loadingMessageIndex: nextIndex })
+    }, 4000)
+  }
+
+  search = async () => {
```

Comment:
don't you think `generate()` would be more semantically accurate here ? . I can see you tried to keep the name consistent with Unsplash provider, but we're literally generating the image here not searching.